compiler/libsolidity/analysis/TypeChecker.cpp

```diff
@@ -643,6 +643,18 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 			m_errorReporter.typeError(_function.location(), "\"virtual\" and \"private\" cannot be used together.");
 	}
 
+	if (_function.externalMsg() || _function.internalMsg()) {
+		if (_function.externalMsg() && _function.internalMsg()) {
+			m_errorReporter.typeError(_function.location(), R"("internalMsg" and "externalMsg" cannot be used together.)");
+		}
+		if (!_function.isPublic()) {
+			m_errorReporter.typeError(_function.location(), R"(Private/internal function can't be marked as internalMsg/externalMsg.)");
+		}
+		if (_function.isReceive() || _function.isFallback() || _function.isOnBounce() || _function.isOnTickTock()) {
+			m_errorReporter.typeError(_function.location(), R"(receiver, fallback, onBounce and onTickTock functions can't be marked as internalMsg/externalMsg.)");
```

Comment:
I would expect a compiler to silently ignore this modifier for the aforementioned functions since they naturally behave this way.